signalfx-codahale/src/main/java/com/codahale/metrics/ResettingExponentiallyDecayingReservoir.java

```diff
@@ -136,14 +136,14 @@ private void rescaleIfNeeded() {
     @Override
     public Snapshot getSnapshot() {
         rescaleIfNeeded();
-        lockForRegularUsage();
+        lockForRescale();
```

Comment:
Actually, to avoid taking and releasing the lock twice, with someone else potentially taking it in the middle, can you `lockForRescale()` before `rescaleIfNeeded()`? `rescale()` will take the (re-entrant) lock, which will then be a no-op since we already own it.